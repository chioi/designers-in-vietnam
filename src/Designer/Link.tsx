import React from "react";

interface ILinkProps {
  url?: string;
}

const Link: React.FC<ILinkProps> = ({ url, children }) => {
  return (
    <a className="designer-link" href={url}>
      <div className="display-flex align-items-center">{children}</div>
    </a>
  );
};

export default Link;
