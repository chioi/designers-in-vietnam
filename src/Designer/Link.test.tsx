import React from "react";
import { create, ReactTestRenderer } from "react-test-renderer";
import Link from "./Link";

describe("When it gets created", () => {
  let component: ReactTestRenderer;

  beforeAll(() => {
    component = create(<Link url="www.test.url">Test Link</Link>);
  });

  it("matches the snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
