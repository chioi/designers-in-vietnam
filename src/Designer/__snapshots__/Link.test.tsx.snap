// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`When it gets created matches the snapshot 1`] = `
<a
  className="designer-link"
  href="www.test.url"
>
  <div
    className="display-flex align-items-center"
  >
    Test Link
  </div>
</a>
`;
